### ISSUE TITLE
GDK_IS_DEVICE (device) failed when draggable widget is moved

### ISSUE THREAD

**Issue by cedlemo** [ruby-gnome2/ruby-gnome2#769]
I have some widgets that are configured to be draggable. Now each time I click and mouve the cursor  those errors messages are displayed:

```
/home/cedlemo/.gem/ruby/2.3.0/gems/gobject-introspection-3.0.9/lib/gobject-introspection/loader.rb: line 554
   Gdk-CRITICAL **:gdk_device_get_source: assertion 'GDK_IS_DEVICE (device)' failed
/home/cedlemo/.gem/ruby/2.3.0/gems/gobject-introspection-3.0.9/lib/gobject-introspection/loader.rb: line 554
   Gdk-CRITICAL **:gdk_device_get_associated_device: assertion 'GDK_IS_DEVICE (device)' failed

```

The same problem appears when I try the drag and drop functionnalities on the **gtk3/samples/misc/dnd.rb** sample.

FYI: I have gtk 3.20.4 on archlinux x86_64.

**Comment by kou** [comment]
It's a GTK+ bug. https://git.gnome.org/browse/gtk+/commit/?id=358eec297204b438809692a24cc3649658dbab5a is a fix.
The fix is included in GTK+ 3.21.1.

**Comment by cedlemo** [comment]
Thanks, I will wait for the update on my system.
